packages/vscode-extension/src/devices/AndroidEmulatorDevice.ts

```diff
@@ -371,6 +376,48 @@ export class AndroidEmulatorDevice extends DeviceBase {
     ]);
   }
 
+  async mirrorNativeLogsDispose() {
+    this.nativeLogsOutputChannel?.dispose();
+    this.nativeLogsProcess?.kill();
+  }
+
+  async mirrorNativeLogs(build: AndroidBuildResult) {
+    const startTime = strftime("%F %T.000", new Date());
+
+    const extractPidFromLogcat = async () =>
+      new Promise<string>((resolve, reject) => {
+        const process = exec(ADB_PATH, ["logcat"]);
+
+        lineReader(process).onLineRead((line) => {
+          const regex = new RegExp(`Start proc ([0-9]{4}):${build.packageName}`);
+
+          if (regex.test(line)) {
+            const groups = regex.exec(line);
+            const pid = groups?.[1];
+            process.kill();
+
+            if (pid) {
+              resolve(pid);
+            } else {
+              reject(new Error("PID not found"));
+            }
+          }
+        });
+      });
+
+    const nativeLogsOutputChannel = window.createOutputChannel("Radon IDE (Android Native Logs)", {
```

Comment:
we should just assign it to `this.nativeLogsOutputChannel` immediately. In case anything later throws, we want to dispose things properly. We can also make it a bit "safer" and only create it when the variable is not already assigned – this way in case this method is called twice, it will still work properly (otherwise to make things safe we should "dispose" previously set variable and create new one, but this doesn't make sense here because those would be the same instaces)

Reply:
Changed in: https://github.com/software-mansion/radon-ide/pull/734/commits/bfa2d8cf60e5d5278b4150daee0dd02cb93e3778